### PR TITLE
FIX Dont use --ignore-platform-reqs

### PR DESCRIPTION
**Pull request by dhensby**
We should never use `--ignore-platform-reqs` with composer. Whilst it appeared to be helpful at the time it's actually making things worse.

Cow needs to define the platform using [`composer config`](https://getcomposer.org/doc/06-config.md#platform) rather than outright ignoring the platform (which is effectively an `--install-highest` flag).

The killer problem with `--ignore-platform-reqs` is that even if we use `composer config` to define the platform (hint: we do) the flag means composer ignores that *as well as* our actual platform constraints.

All in all, it results in creating archives which only run on the highest of all our valid dependencies (PHP 7.2) rather than the lowest (5.6.0)

tl;dr `--ignore-platform-reqs` is evil, we must destroy it

### CODE REVIEW

**Comment by dhensby** [comment]
> I like the fix in #95 better, but with #95 merged we can also drop the option.

I'm not sure that they are related @tractorcow ?

>Does this need any extra config anywhere to set the PHP version in config?

@robbieaverill at the moment we already set it when we do create-project and installs, so we shouldn't need to add anything else. What we could do is be more certain about our other php module dependencies (like `php-intl`) but that's a "nice to have" piece of work as it would mean resolving all the nested dependencies and matching their platform requirements.

**Comment by tractorcow** [comment]
I'm referring to this.

image

**Comment by dhensby** [comment]
Ah, right, yes indeedy

**Comment by tractorcow** [comment]
Be careful, because the initial install should still ignore platform-deps if it has `--no-install`. This is because we rely on that initial clone of the project before we can even read `.cow.json`, which will come into play in the `composer install` step.

I would also suggest to have a set of `$customConfig['platform.<someoption>'] = ['<some/value>'];` assignments. These should probably come from a config section in `.cow.json` file, so that you can specify preferred install platform targeted versions for each project.

So to summarise:

 - Remove `--ignore-platform-deps` except in initial shallow clone / install for bootstrapping the project folder.
 - Ensure you set platform dependencies prior to any other `composer update / install` dependencies.

**Comment by dnsl48** [comment]
That's good to know, thank you. 
So, since we need to leave `--ignore-platform-deps` active for `create-project --no-install`, then it's actually broken in this PR, as `src/Utility/Composer.php` now issues a user warning for that flag and doesn't actually pass it to the command line.

**Comment by dnsl48** [comment]
I don't think we need to break the API of `Utility\Composer` this way.
Leaving the parameter `$ignorePlatform` to exist, but ignoring it is a part of business logic of the application.
A Utility component should refrain from making such decisions and leave it for the upper levels of the application where the business logic is defined.
